go-controller/pkg/retry/obj_retry.go

```diff
@@ -636,10 +644,12 @@ func (r *RetryFramework) WatchResourceFiltered(namespaceForFilteredHandler strin
 					return
 				}
 				klog.V(5).Infof("Delete event received for %s %s", r.ResourceHandler.ObjType, key)
-				// If object is in terminal state, we would have already deleted it during update.
-				// No reason to attempt to delete it here again.
-				if r.ResourceHandler.IsObjectInTerminalState(obj) {
-					klog.Infof("Ignoring delete event for resource in terminal state %s %s",
+				// If object is in terminal state, check if we have already processed it in a previous update.
+				// We cannot blindly handle multiple delete operations for the same pod currently. There can be races
+				// where other pod handlers are removing IP addresses from address sets when they shouldn't be, etc.
+				// See: https://github.com/ovn-org/ovn-kubernetes/pull/3318#issuecomment-1349804450
+				if r.ResourceHandler.IsObjectInTerminalState(obj) && r.ResourceHandler.DeleteAlreadyProcessed(obj) {
```

Comment:
won't this skip delete event for all handlers after PodHandler? 
We know handlers are ordered by priority, so PodHandler will be first, then e.g. PeerPodHandler. When PeerPodHandler executed this part, `DeleteAlreadyProcessed` will be true, but the event was only processed for the first PodHandler?

Reply:
should the delete events be processed in reverse order of priority?